### PR TITLE
Optional React class components lesson

### PR DESCRIPTION
**Pull request by alismith-dev**
Moves existing class components content into an optional lesson solely based on classes. The state lesson had to be re-worked a bit to make sense in the new context.

This frees us up to remove existing content from the main lessons to teach "Hooks-first".

[Rendered version for ease-of-review](https://github.com/CodeYourFuture/syllabus/blob/9e4deae733eff623d80a7553c339966e6b615eb4/react/classes-optional/lesson.md)

### CODE REVIEW

File: react/classes-optional/mentors.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Mentor's Notes - React Class Components
+
+- Class components section
+  - Students often get tripped on converting between functional & class components
```
Comment (zubhav) [inline react/classes-optional/mentors.md:4]: I did a blog post a while ago about converting Classes to function components. I'll share it here incase there is any reason to incorporate it as a link.

https://dev.to/subdeveloper/migrating-from-class-components-to-react-hooks-4lp3

**Comment by zubhav** [comment]
Looks ready to merge to me

File: react/classes-optional/mentors.md (hunk 1 above)
Comment (alismith-dev) [reply react/classes-optional/mentors.md:4]: Thanks I added the link to the further reading at the bottom of the lesson.

**Comment by alismith-dev** [comment]
Thanks @zubhav. Merging.

**Pull request merged**